Homework 3

MegaLexa11 opened the pull request:
Homework 3 done

MegaLexa11 committed:
Added homework_3 tasks

MegaLexa11 committed:
Added getting page with index

avchirkin commented on Homework_3/DynamicArrayProgram/Program.cs:
```diff
@@ -0,0 +1,26 @@
+﻿
+using DynamicArrayProgram;
+
+DynamicArray<string> IntArr =  new();
```
Локальные переменные именуются с маленькой буквы

avchirkin commented on Homework_3/PaginatedArrayProgram/PaginatedArray.cs:
```diff
@@ -0,0 +1,71 @@
+
+        private int _pageSize = pageSize;
+        
+        public void addVal(T value, int place)
```
Названия методов именуются с большой буквы

avchirkin commented on Homework_3/PaginatedArrayProgram/PaginatedArray.cs:
```diff
@@ -0,0 +1,71 @@
+            {
+                _items[place] = value;
+            }
+            // Хотя, по сути, выдаст ту же ошибку, даже если не обрабатывать
```
Тогда исключение излишне оборачивать в try/catch, если ты не хочешь дать пользователю больше информации

avchirkin commented on Homework_3/PaginatedArrayProgram/PaginatedArray.cs:
```diff
@@ -0,0 +1,71 @@
+        {
+            get
+            {
+                int startIndex = (index - 1) * _pageSize;
```
Тут разрыв шаблона получается. Если мы попробуем получить доступ к странице с индексом 0, то упадём с ошибкой. Семантика PaginatedArray должна совпадать с обычным массивом, поэтому индексация страниц должна быть с 0

avchirkin reviewed (changes requested):
Небольшие замечания

MegaLexa11 committed:
Fixed all the problems (probably)

Исправлены замечания (во втором ДЗ, думаю, вышло сделать код более декларативным). Также все сделано в рамках данной ветки, так как пока что структура веток в репозитории совсем сомнительная получилась, хочу это замерджить, а потом уже по каждому дз нормально ветки создавать

avchirkin reviewed (approved):
Молодец, алгоритм работает
